executor/distsql.go

```diff
@@ -233,6 +237,14 @@ func (e *TableReaderExecutor) Open(ctx context.Context) error {
 	span, ctx := startSpanFollowsContext(ctx, "executor.TableReader.Open")
 	defer span.Finish()
 
+	var err error
+	if e.haveCorCol {
+		e.dagPB.Executors, _, err = constructDistExec(e.ctx, e.plans)
```

Comment:
Why we need to re-constructDistExec when executing?

Reply:
Because the value of correlated column is changed every time when we call `Open` to start a new execution.

Reply:
So, we do not need to `constructDistExec` in this func?
``` go
func (b *executorBuilder) constructDAGReq(plans []plan.PhysicalPlan) (dagReq *tipb.DAGRequest, streaming bool, err error) {
	dagReq = &tipb.DAGRequest{}
	dagReq.StartTs = b.getStartTS()
	dagReq.TimeZoneOffset = timeZoneOffset(b.ctx)
	sc := b.ctx.GetSessionVars().StmtCtx
	dagReq.Flags = statementContextToFlags(sc)
	dagReq.Executors, streaming, err = constructDistExec(b.ctx, plans)
	return dagReq, streaming, errors.Trace(err)
}
```

Reply:
We'd better do this in the func. Otherwise we need add a flag to tell that `We don't have correlated column and we've construct dist executors already.` It's possible that the data source executor open multiple times but it don't have cor col.